libs/vkd3d/state.c

```diff
@@ -2626,12 +2626,35 @@ static unsigned int vkd3d_get_rt_format_swizzle(const struct vkd3d_format *forma
 
 STATIC_ASSERT(sizeof(struct vkd3d_shader_transform_feedback_element) == sizeof(D3D12_SO_DECLARATION_ENTRY));
 
+static VkImageAspectFlags vk_dsv_format_to_aspects(VkFormat vk_format)
```

Comment:
Do we know if it's legal to specify something like `R32_UINT` or `R32_FLOAT_X8X24_TYPELESS` as pipeline DSV formats?

If not, `vkd3d_format` already knows the exact aspect mask, so this could potentially be simplified (we'd need to fix the `DXGI_FORMAT_R24_UNORM_X8_TYPELESS` mapping + fallback logic, though).

Reply:
I suppose we could just look at the dsv->resource's aspect mask instead.

Reply:
Should be fixed now.